### PR TITLE
kernel: bump 6.6 to 6.6.66

### PR DESCRIPTION
**Pull request by graysky2**
Update patch set for new release and add required kernel option CONFIG_ZRAM_TRACK_ENTRY_ACTIME to generic config

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.66

Manually rebased:
        bcm27xx/patches-6.6/950-0092-MMC-added-alternative-MMC-driver.patch
        bcm53xx/patches-6.6/180-usb-xhci-add-support-for-performing-fake-doorbell.patch
        starfive/patches-6.6/1000-serial-8250_dw-Add-starfive-jh7100-hsuart-compatible.patch

Removed upstreamed:
        bcm27xx/patches-6.6/950-0029-vc4_hdmi-Avoid-log-spam-for-audio-start-failure.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.66&id=e0388a95736abd1f5f5a94221dd1ac24eacbd4d7

Build system: x86/64
Build-tested: bcm27xx/bcm2712, flogic/glinet_gl-mt6000, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64
Run-tested: bcm27xx/bcm2712, flogic/glinet_gl-mt6000, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64

### CODE REVIEW

File: target/linux/generic/config-6.6 [hunk 1]
```diff
@@ -7548,5 +7548,6 @@ CONFIG_ZONE_DMA=y
 # CONFIG_ZRAM_MEMORY_TRACKING is not set
 # CONFIG_ZRAM_MULTI_COMP is not set
 # CONFIG_ZSMALLOC is not set
+# CONFIG_ZRAM_TRACK_ENTRY_ACTIME is not set
```
Comment (namiltd) [inline target/linux/generic/config-6.6:7551]: It should be one line above

File: target/linux/starfive/patches-6.6/1000-serial-8250_dw-Add-starfive-jh7100-hsuart-compatible.patch [hunk 2]
```diff
@@ -15,11 +15,11 @@ Signed-off-by: Emil Renner Berthing <kernel@esmil.dk>
...
 --- a/drivers/tty/serial/8250/8250_dw.c
 +++ b/drivers/tty/serial/8250/8250_dw.c
-@@ -802,6 +802,7 @@ static const struct of_device_id dw8250_
- 	{ .compatible = "cavium,octeon-3860-uart", .data = &dw8250_octeon_3860_data },
+@@ -803,6 +803,7 @@ static const struct of_device_id dw8250_
  	{ .compatible = "marvell,armada-38x-uart", .data = &dw8250_armada_38x_data },
  	{ .compatible = "renesas,rzn1-uart", .data = &dw8250_renesas_rzn1_data },
+ 	{ .compatible = "sophgo,sg2044-uart", .data = &dw8250_skip_set_rate_data },
 +	{ .compatible = "starfive,jh7100-hsuart", .data = &dw8250_starfive_jh7100_data },
- 	{ .compatible = "starfive,jh7100-uart", .data = &dw8250_starfive_jh7100_data },
```
Comment (namiltd) [inline target/linux/starfive/patches-6.6/1000-serial-8250_dw-Add-starfive-jh7100-hsuart-compatible.patch:23]: Change to:
```
{ .compatible = "starfive,jh7100-uart", .data = &dw8250_skip_set_rate_data },
```

File: target/linux/starfive/patches-6.6/1000-serial-8250_dw-Add-starfive-jh7100-hsuart-compatible.patch [hunk 3]
```diff
@@ -15,11 +15,11 @@ Signed-off-by: Emil Renner Berthing <kernel@esmil.dk>
 
 --- a/drivers/tty/serial/8250/8250_dw.c
 +++ b/drivers/tty/serial/8250/8250_dw.c
-@@ -802,6 +802,7 @@ static const struct of_device_id dw8250_
- 	{ .compatible = "cavium,octeon-3860-uart", .data = &dw8250_octeon_3860_data },
+@@ -803,6 +803,7 @@ static const struct of_device_id dw8250_
  	{ .compatible = "marvell,armada-38x-uart", .data = &dw8250_armada_38x_data },
  	{ .compatible = "renesas,rzn1-uart", .data = &dw8250_renesas_rzn1_data },
+ 	{ .compatible = "sophgo,sg2044-uart", .data = &dw8250_skip_set_rate_data },
 +	{ .compatible = "starfive,jh7100-hsuart", .data = &dw8250_starfive_jh7100_data },
```
Comment (namiltd) [inline target/linux/starfive/patches-6.6/1000-serial-8250_dw-Add-starfive-jh7100-hsuart-compatible.patch:22]: Change to:
```
{ .compatible = "starfive,jh7100-hsuart", .data = &dw8250_skip_set_rate_data },
```

Comment (graysky2) [reply target/linux/starfive/patches-6.6/1000-serial-8250_dw-Add-starfive-jh7100-hsuart-compatible.patch]: Done, thanks for the review!

**Review by namiltd — APPROVED** [review]

**Comment by rsalvaterra** [comment]
Please hold on. There's a [known bug](https://lore.kernel.org/lkml/CALjTZvYKHWrD5m+RXimjxODvpFPw7Cq_EOEuzRi1PZT9_JxF+g@mail.gmail.com/) affecting at least some tg3 Ethernet interfaces. Whilst I reported it in 6.13-rc1, the relevant commit has been [backported to stable](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/net/ethernet/broadcom?h=linux-6.6.y&id=d0f06266562b92729ce3ad3c642f93397fbe29b1). I have a patch that fixes the issue for me, it has been sent to me by the author off-list, and I'm attaching it here. It's been signed off, so it's legally safe to apply.

[0001-tg3-Fix-DMA-allocations-on-57766-devices.patch.zip](https://github.com/user-attachments/files/18140238/0001-tg3-Fix-DMA-allocations-on-57766-devices.patch.zip)

**Comment by graysky2** [comment]
@rsalvaterra - I will add it to `pending-6.6`

**Comment by namiltd** [comment]
I wouldn't implement this fix in this PR, but in a separate one. It would need to be tested on a 6.6.x kernel first. Does it even apply to that?

**Comment by graysky2** [comment]
OK, I did not include the patch.  Here it is after passing through `update_kernel.sh`:
[791-tg3-Fix-DMA-allocations-on-57766-devices.patch.gz](https://github.com/user-attachments/files/18140622/791-tg3-Fix-DMA-allocations-on-57766-devices.patch.gz)

**Comment by rsalvaterra** [comment]
I'm AFK, so please do it, if you can. Feel free to add my Tested-by.

A domingo, 15/12/2024, 13:53, graysky ***@***.***> escreveu:

**Comment by rsalvaterra** [comment]
Additionally, the patch is also needed for 6.1, I'm afraid.

A domingo, 15/12/2024, 14:09, Rui Salvaterra ***@***.***>
escreveu:

**Comment by graysky2** [comment]
@rsalvaterra - https://github.com/openwrt/openwrt/pull/17282

**Comment by namiltd** [comment]
Duplicated # CONFIG_ZRAM_TRACK_ENTRY_ACTIME is not set
in target/linux/generic/config-6.6

**Comment by namiltd** [comment]
I think it would be enough to mention at the end (without these sub-items):
Added the required kernel setting CONFIG_ZRAM_TRACK_ENTRY_ACTIME to the generic config

**Comment by namiltd** [comment]
Because the main topic is changing the kernel version please change description to:
Update patch set for new release and add required kernel option CONFIG_ZRAM_TRACK_ENTRY_ACTIME to generic config.

**Comment by namiltd** [comment]
The title was ok, this text was supposed to be in the description

**Review by namiltd — APPROVED** [review]

**Pull request merged**